app/services/linking_stat_collator.rb

```diff
@@ -0,0 +1,72 @@
+class LinkingStatCollator < ApplicationService
+  def initialize(date_from, date_to)
+    @date_from = date_from
+    @date_to = date_to
+  end
+
+  NUMBER_OF_PREVIOUS_RANGES = 10
+
+  attr_reader :date_from, :date_to
+
+  def call
+    {
+      current_range:,
+      previous_ranges:,
+    }
+  end
+
+  # We are comparing dates to timestamps. By default a date is treated as being the very
+  # start of the day, so to include all timestamps occurring on that day we need to set
+  # the cutoff as being the start of the _following_ day.
+  def date_to_cutoff
+    date_to + 1.day
+  end
+
+  def current_range
+    {
+      date_from:,
+      date_to:,
+      linked:,
+      unlinked:,
+      unlink_reason_codes:,
+      other_unlink_reasons:,
+    }
+  end
+
+  def linked
+    LaaReference.where(created_at: date_from..date_to_cutoff).count
+  end
+
+  def unlinked
+    LaaReference.where(created_at: date_from..date_to_cutoff, linked: false).count
+  end
+
+  def unlink_reason_codes
+    LaaReference.where(created_at: date_from..date_to_cutoff, linked: false).pluck(:unlink_reason_code).tally
+  end
+
+  def other_unlink_reasons
+    LaaReference.where(created_at: date_from..date_to_cutoff, linked: false)
+                .where.not(unlink_other_reason_text: nil).pluck(:unlink_other_reason_text)
+  end
```

Comment:
What about reusing `unlinked` ?

```
  def unlinked
    LaaReference.where(created_at: date_from..date_to_cutoff, linked: false)
  end

  def unlink_reason_codes
    unlinked.pluck(:unlink_reason_code).tally
  end

  def other_unlink_reasons
    unlinked.where.not(unlink_other_reason_text: nil).pluck(:unlink_other_reason_text)
  end
```

Of course, you should use `unlinked.count`, which is probably clearer.

You may want to remove `.count` from `linked` for consistency.